stdlib/number.gr

```diff
@@ -493,3 +952,107 @@ export let cos = (radians: Number) => {
     sin(pi / 2 + radians)
   }
 }
+
+/**
+ * Computes the tangent of a number (in radians) using Chebyshev polynomials.
+ *
+ * @param radians: The input in radians
+ * @returns The computed tangent
+ *
+ * @since v0.5.4
+ */
+export let tan = (radians: Number) => {
+  if (isNaN(radians) || isInfinite(radians)) {
+    nan
+  } else {
+    sin(radians) / cos(radians)
+  }
+}
+
+// Math.gamma implemented using the stirling's approximation
+// https://en.wikipedia.org/wiki/Lanczos_approximation
+/**
+ * Computes the gamma function of a value using
+ *
+ * @param z: The value to compute the gamma of
+ * @returns The gamma of the given value
+ *
+ * @since v0.5.4
+ */
+export let rec gamma = z => {
+  if (z == 0) {
+    fail "Gamma of 0 is undefined"
+  } else if (isInteger(z) && z > 0) {
+    let mut output = 1
+    for (let mut i = 1; i < z; i += 1) {
+      output *= i
+    }
+    output
+  } else {
+    let mut z = z
+    let g = 7
+    let c = [>
+      0.99999999999980993,
+      676.5203681218851,
+      -1259.1392167224028,
+      771.32342877765313,
+      -176.61502916214059,
+      12.507343278686905,
+      -0.13857109526572012,
+      9.9843695780195716e-6,
+      1.5056327351493116e-7,
+    ]
+    let mut output = 0
+    if (z < 0.5) {
+      output = pi / (sin(pi * z) * gamma(1 - z))
+    } else if (z == 0.5) {
+      // Handle this case separately because it is out of the domain of Number.pow when calculating
+      output = 1.7724538509055159
+    } else {
+      z -= 1
+      let mut x = c[0]
+      for (let mut i = 1; i < g + 2; i += 1) {
+        x += c[i] / (z + i)
+      }
+
+      let t = z + g + 0.5
+      output = sqrt(2 * pi) * pow(t, z + 0.5) * exp(t * -1) * x
+    }
+    if (abs(output) == infinity) infinity else output
+  }
+}
+
+/**
+ * Computes the factorial of a given number
```

Comment:
```suggestion
 * Computes the product of consecutive integers in the given number.
```
Is this correct? Just quickly scanned wikipedia

Reply:
As discussed in discord, I dont think this is really the correct defintion as we support factorial on decimals through the gamma function, and changing the word integers to numbers creates questions like for instance i could make a very clean arugment that `0.01` exists in `5` and so on along with argument of what numbers exist in a decimal. Maybe we do:
```
Computes the product of consecutive integers in the given number.
When given a foat it uses the gamma function as an aproximation.
```

Reply:
```suggestion
 * Computes the product of consecutive integers for an integer input and computes the gamma function for non-integer inputs.
```

Reply:
@ospencer While we are here, we should probably add `Fails if the input is a  negative number.` as a separate sentence
```suggestion
 * Computes the product of consecutive integers for an integer input and computes the gamma function for non-integer inputs.
 * Fails if the input is a negative number.
```